pkg/runtime/config.go

```diff
@@ -346,6 +348,18 @@ func (c *Config) toInternal() (*internalConfig, error) {
 		}
 	}
 
+	if c.ApplicationHealthPort != "" {
+		intc.appConnectionConfig.HealthPort, err = strconv.Atoi(c.ApplicationHealthPort)
```

Comment:
Please add unit tests for these two blocks.

We should also use `intc.appConnectionConfig.Port` to assign to `intc.appConnectionConfig.HealthPort`rather than re-parsing the string.